### PR TITLE
Improve valgrind suppression file generation

### PR DESCRIPTION
**Pull request by jgfouca**
Include an executable that is linked the same way most catch exes are linked.

## Motivation
This will add some suppressions that will help to keep innocuous valgrind errors down.

## Testing
Scream ctests with valgrind enabled.

### CODE REVIEW

File: valgrind_support/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,10 @@
+include(EkatCreateUnitTest)
 
 add_executable(serial_hw serial_hw.cxx)
 add_executable(mpi_hw mpi_hw.cxx)
 
+EkatCreateUnitTest(mpi_valg_catch_test mpi_catch_test.cpp)
```
Comment (bartgol) [inline valgrind_support/CMakeLists.txt:6]: I think you might have forgotten to include this file in the commit maybe?

Comment (jgfouca) [reply valgrind_support/CMakeLists.txt:6]: Good catch! Fixed.

**Review by bartgol — APPROVED** [review]
Looks good. Do you think it makes sense to eliminate `mpi_hw` exec at this point? The new one seems to create a supp file that is a superset of the old one, so maybe the old one is pointless?

**Comment by jgfouca** [comment]
@bartgol , interestingly, for the version of openmpi that I am going to set mappy to for scream, both are needed.

**Pull request merged**